### ISSUE TITLE
[FeatureRequest] Merge kfold types

### ISSUE THREAD

**Issue by mucki123** [delira-dev/delira#86]
**Description**
What should be added/changed?
Do a KFold_predict where u can turn on/off straitfied

**Feature History**
What have you tried so far?
maybe just delte all labels something like this:
   metrics_val = {}
        outputs = {}
        

        if num_splits is None:
            num_splits = 10
            logger.warning("num_splits not defined, using default value of \
                                       10 splits instead ")

        if isinstance(data.dataset, BaseLazyDataset):
            logger.warning("A lazy dataset is given for stratified kfold. \
                                   Iterating over the dataset to extract labels for \
                                   stratification may be a massive overhead")

        split_idxs = list(range(len(data.dataset)))
        

        fold = KFold(n_splits=num_splits, shuffle=shuffle,
                               random_state=random_seed)

        for idx, (_train_idxs, test_idxs) in enumerate(fold.split(split_idxs,
                                                                  split_labels)):
            # extract data from single manager
            _train_data = data.get_subset(_train_idxs)
            _split_idxs = list(range(len(_train_data.dataset)))
            

            val_fold = ShuffleSplit(n_splits=1,
                                              test_size=split_val,
                                              random_state=random_seed)

            for train_idxs, val_idx in val_fold.split(_split_idxs):
                train_data = _train_data.get_subset(train_idxs)
                val_data = _train_data.get_subset(val_idx)

            test_data = data.get_subset(test_idxs)

            # update manager behavior for train and test case
            train_data.update_state_from_dict(train_kwargs)
            val_data.update_state_from_dict(test_kwargs)
            test_data.update_state_from_dict(test_kwargs)
            model = self.run(train_data, val_data,
                             num_epochs=num_epochs,
                             fold=idx,
                             **kwargs)

            _outputs, _metrics_val = self.test(
                self.params, model, test_data)

            outputs[str(idx)] = _outputs
            metrics_val[str(idx)] = _metrics_val

        return outputs, metrics_val
**Proposal**
How could the feature be implemented? 
*Are you able/willing to implement the feature yourself (with some guidance from us)?

**Additional context**
Add any other context about the feature request here.

**Comment by justusschock** [comment]
Seems to be reasonable but low priority (only to unify similar code). If you're able to do this yourself, I'd review. Otherwise this issue will done at any (undetermined) time in the future.

Probably we should first merge `kfold` and `stratified_kfold` and afterwards look for a possibility to call the new unified `kfold` inside the `kfold_predict` to further remove duplicate code. Do you think, you can do this? If you encounter any problems, just tag me and I'll try to figure them out.

**Comment by mucki123** [comment]
Hi Justus, 
ich hab einfach alle Labels rausgelöscht, weil ich ja eine Segementierung unter Tensorflow mache. Auch im Return der Test-Function scheint jetzt ganz gut durchzulaufen

**Comment by justusschock** [comment]
Dann verstehe ich nicht, was du am Framework ändern willst

**Comment by mucki123** [comment]
Am besten wäre es ja, wenn man einfach nen Modus wählen könnte, anstatt das einfach alles rauszulöschen, bzw wie du meintest mit dem mergen.Hat natürlich keine hohe Prio, aber ich sollte ein Issue aufmachen
